src/map/script.c

```diff
@@ -18954,6 +18954,19 @@ static BUILDIN(setunitdata)
 	case UDT_ELELEVEL:
 		setunitdata_check_bounds(4, 0, CHAR_MAX);
 		break;
+	case UDT_GROUP:
+	{
+		struct unit_data *ud = unit->bl2ud2(bl);
+		if (ud == NULL) {
+			ShowError("buildin_setunitdata: ud is NULL!\n");
+			script_pushint(st, 0);
+			return false;
+		}
+		ud->groupId = script_getnum(st, 4);
```

Comment:
why is this check done inside the check_bound area ?
isn't this should be done 6 times repeatedly, inside the switch bl->type ?

and is the title ID allow negative input ? if it is, here should be 
`setunitdata_check_bounds(4, 0, INT_MAX);`

Reply:
actually i not sure is it allow negative values. this option is client side configured. normally probably from 0 to INT_MAX but can be something else. because this i not sure is limiting here make sense

Reply:
sorry it should be `setunitdata_check_min(4, 0);`

how does it work ? doesn't seem to have any visual effect in-game
```c
prontera,155,185,5	script	kjhsdff	1_F_MARIA,{
	.@mobgid = monster("this", -1,-1, "--ja--", PORING,1 );
	setunitdata .@mobgid, UDT_GROUP, 1000;
	setunitdata getnpcid(), UDT_GROUP, 1000;
}
```

Reply:
@AnnieRuru groups + titles is client feature for displaying a name prefix and an icon for npcs and mobs, it looks like this 

![](https://media.discordapp.net/attachments/359827876851482634/566735939896999936/unknown.png)

Reply:
i set limits from 0 to INT_MAX, if need in future updated can be changed